src/js/elements/menu.js

```diff
@@ -41,39 +41,44 @@ function subindicatorsInIndicator(indicator) {
 
 // TODO this entire file needs to be refactored to use thhe observer pattern
 export function loadMenu(dataMapperMenu, data, subindicatorCallback) {
-    function addSubIndicators(wrapper, category, subcategory, indicator, groups, indicators, choropleth_method, indicatorId) {
+    function addSubIndicators(wrapper, category, subcategory, indicator, groups, indicators, indicatorDetail) {
 
         $(".data-category__h3", wrapper).remove();
         $(".data-category__h4", wrapper).remove();
 
         if (groups !== null && typeof groups.subindicators !== 'undefined') {
-            groups.subindicators.forEach((subindicator) => {
-                const newSubIndicatorElement = indicatorItemTemplate.cloneNode(true);
-                $(".truncate", newSubIndicatorElement).text(subindicator);
-                $(newSubIndicatorElement).attr('title', subindicator);
-
-                wrapper.append(newSubIndicatorElement);
 
-                const parents = {
-                    category: category,
-                    subcategory: subcategory,
-                    indicator: indicator
+            groups.subindicators.forEach((subindicator) => {
+                let display = checkIfSubIndicatorHasData(subindicator, indicatorDetail);
```

Comment:
to be clearer, this should be called something like `subindicatorHasData` or just `hasData`

Reply:
Agreed, I renamed and moved the function from utils to menu.js